rdkPlugins/Logging/source/FileSink.cpp

```diff
@@ -0,0 +1,282 @@
+        {
+            mFileSizeLimit = mContainerConfig->rdk_plugins->logging->data->file_options->limit;
+            // Set to -1 in config for "unlimited"
+            if (mFileSizeLimit < 0)
+            {
+                mFileSizeLimit = SSIZE_MAX;
+            }
+        }
+        else
+        {
+            AI_LOG_INFO("No file size limit size for container log - setting to unlimited");
+            mFileSizeLimit = SSIZE_MAX;
+        }
+    }
+
+    mOutputFileFd = openFile(mOutputFilePath);
+    if (mOutputFileFd < 0)
+    {
+        // Couldn't open our output file, send to /dev/null to avoid blocking
+        AI_LOG_SYS_ERROR(errno, "Failed to open container logfile - sending to /dev/null");
+        if (mDevNullFd > 0)
+        {
+            mOutputFileFd = mDevNullFd;
+        }
+    }
+
+    AI_LOG_FN_EXIT();
+}
+
+FileSink::~FileSink()
+{
+    // Close everything we opened
+    if (close(mDevNullFd) < 0)
+    {
+        AI_LOG_SYS_ERROR(errno, "Failed to close /dev/null");
+    }
+
+    if (mOutputFileFd > 0)
+    {
+        if (close(mOutputFileFd) < 0)
+        {
+            AI_LOG_SYS_ERROR(errno, "Failed to close output file");
+        }
+    }
+}
+
+/**
+ * @brief Sets the log options used by the process() method
+ */
+void FileSink::SetLogOptions(const IDobbyRdkLoggingPlugin::LoggingOptions &options)
+{
+    std::lock_guard<std::mutex> locker(mLock);
+    mLoggingOptions = options;
+}
+
+/**
+ * @brief Reads all the available data from the provided fd and writes it to the output
+ * file. Does not attempt to seek the file descriptor back to the start
+ *
+ * If file limit is hit, will send data to /dev/null
+ *
+ * @param[in]   bufferFd    The fd to read from
+ */
+void FileSink::DumpLog(const int bufferFd)
+{
+    std::lock_guard<std::mutex> locker(mLock);
+
+    memset(mBuf, 0, sizeof(mBuf));
+
+    ssize_t ret;
+    ssize_t offset = 0;
+
+    // Read all the data from the provided file descriptor
+    while (true)
+    {
+        ret = read(bufferFd, mBuf, sizeof(mBuf));
+        if (ret <= 0)
+        {
+            break;
+        }
+
+        offset += ret;
+
+        if (offset <= mFileSizeLimit)
+        {
+            // Write to the output file
+            if (write(mOutputFileFd, mBuf, ret) < 0)
+            {
+                AI_LOG_SYS_ERROR(errno, "Write failed");
+            }
+        }
+        else
+        {
+            // Hit the limit, send the data into the void
+            if (!mLimitHit)
+            {
+                AI_LOG_WARN("Logger for container %s has hit maximum size of %zu",
+                            mContainerId.c_str(), mFileSizeLimit);
+            }
+            mLimitHit = true;
+            write(mDevNullFd, mBuf, ret);
+        }
+    }
+
+#if (AI_BUILD_TYPE == AI_DEBUG)
+    // Separate sections of log file for reabability
+    // (useful if we're writing lots of buffer dumps)
+    std::string marker = "---------------------------------------------\n";
```

Comment:
Here checking for mLimitHit would make sense, as if we already exceeded limit then probably we don't wont to write anything more.

Reply:
Done